cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -113,8 +114,25 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
       }
 
       NamespaceConfig config = metadata.getConfig();
-      if (config != null && !Strings.isNullOrEmpty(config.getSchedulerQueueName())) {
-        builder.setSchedulerQueueName(config.getSchedulerQueueName());
+      if (config != null) {
```

Comment:
Why this logic is in the handler? Should the namespace admin take a `NamespaceConfig` and handle it appropriately? E.g. in future, when you add validations of the provided HBase/Hive namespace, it shouldn't be done in here.